### PR TITLE
Create group with needsRotation

### PR DESCRIPTION
**Pull request by giarc3**
- Added `needsRotation` to `GroupCreateOpts`
- Tested default as well as non-default group creation
- Locked tests to staging as that's what the tests assume
- Removed .cargo/config because we no longer need to specify `ironoxide/senv`

Merging into feature branch while catching up with ironoxide 0.14.0
Can depend on ironoxide 0.14.0 once catch-up is done

### CODE REVIEW

File: tests/src/test/scala/ironrust/FullIntegrationTest.scala [hunk 1]
```diff
@@ -294,12 +303,8 @@ class FullIntegrationTest extends DudeSuite with CancelAfterFailure {
       val sdk = IronSdk.initialize(createDeviceContext)
       val groupId = Try(GroupId.validate("not-a-group=ID-that-exists=")).toEither.value
       val resp = Try(sdk.groupGetMetadata(groupId)).toEither
-      resp.leftValue.getMessage should include(
-        """404"""
-      )
-      resp.leftValue.getMessage should include(
-        """Requested resource was not found"""
-      )
+      resp.leftValue.getMessage should include("""404""")
```
Comment (coltfred) [inline tests/src/test/scala/ironrust/FullIntegrationTest.scala]: Nit: You don't need `"""`, just normal `"`

File: tests/src/test/scala/ironrust/FullIntegrationTest.scala [hunk 2]
```diff
@@ -309,7 +314,6 @@ class FullIntegrationTest extends DudeSuite with CancelAfterFailure {
 
       group.getId.getId.length shouldBe 32
       group.getId shouldBe validGroupId
-      group.getName.get.getName shouldBe "a name"
```
Comment (coltfred) [inline tests/src/test/scala/ironrust/FullIntegrationTest.scala]: Why did you remove these?

Comment (giarc3) [reply tests/src/test/scala/ironrust/FullIntegrationTest.scala]: I originally did this because I changed the primary group to use the GroupCreateOpts defaults, which didn't allow specifying a name. So everywhere that we checked that the name was set would be incorrect.
But I decided to undo that change and just add an extra test for the defaults, so I've added all of these assertions back in.

File: tests/src/test/scala/ironrust/FullIntegrationTest.scala (hunk 1 above)
Comment (giarc3) [reply tests/src/test/scala/ironrust/FullIntegrationTest.scala]: I didn't put them there originally, but I now took them out
